jgrapht-core/src/test/java/org/jgrapht/alg/AllDirectedPathsTest.java

```diff
@@ -78,6 +78,24 @@ public void testSmallExampleGraph () {
         assertEquals("Toy network should have correct number of simple paths", 7, allPaths.size());
     }
 
+    public void testTrivialPaths () {
+        AllDirectedPaths<String, DefaultEdge> pathFindingAlg =
+            new AllDirectedPaths<>(toyGraph());
+
+        Set<String> sources = new HashSet<>();
+        sources.add(I1);
+
+        Set<String> targets = new HashSet<>();
+        targets.add(I1);
+        targets.add(A);
+
+        List<GraphPath<String, DefaultEdge>> allPaths =
+            pathFindingAlg.getAllPaths(sources, targets, true, null);
+
+        assertEquals("Toy network should have correct number of trivial simple paths", 2, allPaths.size());
```

Comment:
Since there are only 2 paths, would it be a good idea to explicitly state them and compare with equals? That seems a bit more secure than simply checking the cardinality of the set of paths.